app/forms/candidate_interface/continuous_applications/which_course_are_you_applying_to_step.rb

```diff
@@ -72,11 +79,15 @@ def completed?
     private
 
       def valid_course_choice
-        @valid_course_choice ||= valid?(:course_choice)
+        @valid_course_choice ||= !duplicate_course? && !full_course?
       end
 
       def duplicate_course?
-        !valid_course_choice
+        !valid?(:course_choice)
+      end
+
+      def full_course?
+        course.course_options.available.blank?
```

Comment:
I think we have the course_full? method. Let me confirm after our meeting.

Reply:
Yes, confirmed, this line could be `course.full?`

Reply:
Ok, I can change it, but are you sure it's the right solution? Please take a look [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/2895) and let me know if I should go with `course.full?` CC @JR-G

Reply:
I'll leave it as `.available?` for now, but let me know if I should change it.

Reply:
Just had a read through Steves PR! Due to recent publish/sync changes the view of sites and vacancies has changed. Rather than individual course options being managed in terms of availability, now all associated course options are set to either `vacancies` or `no_vacancies` – we'll never see a combination again.

I just tried these methods on a course that has 10 course options, all with vacancies:

`course.full?` returned `false`
`course.available?` returned `true`

I've set all course options to `no_vacancies` called the methods, and then removed all the course options and called them a final time. On both occasions they returned:

`course.full?` returned `true`
`course.available?` returned `false`

So both are consistent in what they return – I guess `.available` is looking for at least one, so it's prudent but we are a bit more all or nothing now, when it comes to vacancies.

We have tech debt clean up to do as part of the sync fix so for me the most important part is we're consistent in which method we decide to take forward.